src/Core/Image/ImageFormatConfiguration.php

```diff
@@ -51,7 +57,30 @@ public function __construct(ConfigurationInterface $configuration)
 
     public function getGenerationFormats(): array
     {
-        return explode(self::SEPARATOR, $this->configuration->get(self::IMAGE_FORMAT_CONFIGURATION_KEY));
+        // Return formats from cache
+        if (!empty(self::$formatsToGenerate)) {
```

Comment:
Can you please not use static cache, except very specific use cases they are usually a bad practice as they are a pain in the ass to clear and mess with test environments
Since this class is a service using a field cache should have the same behaviour as only one instance of the service will be instanciated in the request process